dev/Saumya/algorithm_helpers.py

```diff
@@ -0,0 +1,321 @@
+                'max_samples': grid_ratio, #default: 1.0
+                'random_state': grid_seed
+                 }],
+
+
+                [{
+                #ExtraTreesClassifier
+                'n_estimators': grid_n_estimator, #default: 10
+                'criterion': grid_criterion, #default: 'gini'
+                'max_depth': grid_max_depth, #default: None
+                'random_state': grid_seed
+                 }],
+
+
+                [{
+                #GradientBoostingClassifier
+                'loss': ['deviance', 'exponential'], #default: ’deviance’
+                'learning_rate': [0.05, 0.1], #default: 0.1, best: 0.1 (This one takes time)
+                'n_estimators': [100, 300], #default: 100, best: 300 (This one takes time)
+                'criterion': ['friedman_mse', 'mse', 'mae'], #default: ”friedman_mse”
+                'max_depth': grid_max_depth, #default: 3   
+                'random_state': grid_seed
+                 }],
+
+
+                [{
+                #RandomForestClassifier
+                'n_estimators': grid_n_estimator, #default: 10
+                'criterion': grid_criterion, #default: 'gini'
+                'max_depth': grid_max_depth, #default: None
+                'oob_score': [True, False], #default: False, best: True (This one takes time)
+                'random_state': grid_seed
+                 }],
+
+                [{    
+                #GaussianProcessClassifier
+                'max_iter_predict': grid_n_estimator, #default: 100
+                'random_state': grid_seed
+                }],
+
+
+                [{
+                #LogisticRegressionCV
+                'fit_intercept': grid_bool, #default: True
+                'penalty': ['l1','l2'],
+                'solver': ['newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga'], #default: 'lbfgs'
+                'random_state': grid_seed
+                 }],
+
+
+                [{
+                #BernoulliNB
+                'alpha': grid_ratio, #default: 1.0
+                 }],
+
+
+                #GaussianNB
+                [{}],
+
+                [{
+                #KNeighborsClassifier
+                'n_neighbors': [1,2,3,4,5,6,7], #default: 5
+                'weights': ['uniform', 'distance'], #default: ‘uniform’
+                'algorithm': ['auto', 'ball_tree', 'kd_tree', 'brute']
+                }],
+
+
+                [{
+                #SVC
+                'kernel': ['linear', 'poly', 'rbf', 'sigmoid'],
+                'C': [1,2,3,4,5], #default: 1.0
+                'gamma': grid_ratio, #edfault: 'auto'
+                'decision_function_shape': ['ovo', 'ovr'], #default: ovr
+                'probability': grid_bool,
+                'random_state': grid_seed
+                 }],
+
+                #LDA
+                [{}],
+
+                #QDA
+                [{}]
+
+            ]
+
+
+
+    start_total = time.perf_counter()
+    for clf, param in zip (vote_est, grid_param):
```

Comment:
I would make `grid_param` into a dict using the same keys as `vote_est` to make it explicit, rather than relying on zip. This will make code updates easier.

Reply:
Yes I later realized that even if the lists are inconsistent *zip* would do its work silently, results would be wrong and I would not even know where the error lies. Thank you for pointing this out!